include/cutlass/gemm/collective/intel_pvc_mma.hpp

```diff
@@ -0,0 +1,353 @@
+    XE_Copy_B gmem_tiled_copy_b;
+  };
+
+  //
+  // Methods
+  //
+
+  CollectiveMma() = default;
+
+  template <class ProblemShape>
+  static constexpr Params
+  to_underlying_arguments(ProblemShape const& problem_shape, Arguments const& args, void* workspace) {
+    (void) workspace;
+
+    auto problem_shape_MNKL = append<4>(problem_shape, 1);
+    auto [M,N,K,L] = problem_shape_MNKL;
+
+    Tensor tensorA = make_tensor(args.ptr_A, make_layout(make_shape(M,K,L), args.dA));
+    Tensor tensorB = make_tensor(args.ptr_B, make_layout(make_shape(K,N,L), args.dB));
+
+    typename Params::XE_Copy_A copyA = make_xe_2d_copy<GmemTiledCopyA>(tensorA);
+    typename Params::XE_Copy_B copyB = make_xe_2d_copy<GmemTiledCopyB>(tensorB);
+    return Params{copyA, copyB};
+  }
+
+  /// Perform a subgroup-scoped matrix multiply-accumulate
+  template <
+    class FrgTensorD,
+    class TensorA,
+    class TensorB,
+    class FrgTensorC,
+    class KTileIterator,
+    class ResidueMNK
+  >
+  CUTLASS_DEVICE void
+  operator() (
+      FrgTensorD &accum,
+      TensorA gA,
+      TensorB gB,
+      FrgTensorC const &src_accum,
+      KTileIterator k_tile_iter, int k_tile_count,
+      ResidueMNK residue_mnk,
+      int thread_idx,
+      char *smem_buf,
+      Params const& mainloop) 
+  {
+    (void)residue_mnk;
+    (void)thread_idx;
+    (void)smem_buf;
+
+    static_assert(is_rmem<FrgTensorD>::value, "D tensor must be rmem resident.");
+    static_assert(is_tuple<typename TensorA::engine_type::iterator::value_type>::value, "A tensor must be a tuple iterator.");
+    static_assert(is_tuple<typename TensorB::engine_type::iterator::value_type>::value, "B tensor must be a tuple iterator.");
+    static_assert(is_rmem<FrgTensorC>::value, "C tensor must be rmem resident.");
+
+    // Tensor to hold input data
+    Tensor tAr = make_tensor<ushort>(Shape<Int<get<0>(TileShape{}) * FragsK>, Int<1>>{});
+    Tensor tBr = make_tensor<ushort>(Shape<Int<FragsK * get<1>(TileShape{}) / FragsN>, Int<FragsN>>{});
+
+    Tensor tAr_view = make_tensor(static_cast<decltype(tAr) &&>(tAr).data(),
+                            Shape<Int<VecA>, Int<FragsM>, Int<FragsK>>{});
+    Tensor tBr_view = make_tensor(static_cast<decltype(tBr) &&>(tBr).data(),
+                            Shape<Int<VecB>, Int<FragsK>, Int<FragsN>>{});
+
+    // Instantiate the M MA object
+    TiledMma tiled_mma;
+
+    //
+    // Mainloop
+    //
+   for (int k_tile = 0, k = 0; k_tile < k_tile_count; ++k_tile, k += DpasK * FragsK)
+   {
+     // Copy gmem to rmem for the first k_tile
+     copy(mainloop.gmem_tiled_copy_a, gA(_,_,k), tAr);
+     copy(mainloop.gmem_tiled_copy_b, gB(_,k/2,_), tBr);
+
+     for (int kl = 0; kl < FragsK; kl++) {
+       cute::gemm(tiled_mma, accum, tAr_view(_, _, kl), tBr_view(_, kl, _), src_accum);
+     }
+   }
+  }
+};
+
+/////////////////////////////////////////////////////////////////////////////////////////////////
+
+// TODO: this code hasn't been investigated yet.
```

Comment:
what do you mean by this comment?

Reply:
This partial specialization of `ComputeMma` is for the case when we have corner cases in the input/output matrices, and we want to add code that considers these possibilities while computing GEMM. I think this approach was followed in `sm_70` whereas for `sm_90` they create a different file for each partial specialization of `ComputeMma` struct. I can remove this from here if need be.

Reply:
If it is not tested, it should not be in this pr. If it need investigation and require further assessment, it should be in separate pr so that it can have its own test and assessment. If it is necessary to be in this pr, the pr, then we need to put this pr as a draft pr and until fixing this investigation we should keeput as  draft